java_src/foam/dao/MapDAO.java

```diff
@@ -50,8 +50,11 @@ public FObject remove(FObject obj) {
   }
 
   public FObject find(Object o) {
-    Object id = getOf().getClass().isInstance(o) || false ? getPrimaryKey().get(o) : o;
-    return getData().get(id);
+    if(getOf().getObjClass().isInstance(o)) {
```

Comment:
See the FOAM style-guide. Should be:

if ( getOf().getObjClass().isInstance(o) ) {